File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/TemperatureMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Temperature
import com.phamtunglam.health_connector_hc_android.pigeon.TemperatureDto

/**
 * Converts a [TemperatureDto] to a Health Connect [Temperature] object.
 */
internal fun TemperatureDto.toHealthConnect(): Temperature = Temperature.celsius(celsius)

/**
 * Converts a Health Connect [Temperature] object to a [TemperatureDto].
 *
 * Uses celsius as the transfer unit for consistency.
 */
internal fun Temperature.toDto(): TemperatureDto = TemperatureDto(
    celsius = inCelsius,
)

/**
 * Converts a temperature value (Double in Celsius) to a [TemperatureDto].
 *
 * Uses celsius as the transfer unit for consistency.
 */
internal fun Double.toTemperatureDto(): TemperatureDto = TemperatureDto(
    celsius = this,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/temperature_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Temperature;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show TemperatureDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Temperature] to [TemperatureDto].
@internal
extension TemperatureToDto on Temperature {
  TemperatureDto toDto() {
    // Uses celsius as the transfer unit for consistency.
    return TemperatureDto(
      celsius: inCelsius,
    );
  }
}

/// Converts [TemperatureDto] to [Temperature].
@internal
extension TemperatureDtoToDomain on TemperatureDto {
  Temperature toDomain() {
    return Temperature.celsius(celsius);
  }
}
